src/test/java/org/jabref/logic/exporter/BibtexDatabaseWriterTest.java

```diff
@@ -289,7 +289,56 @@ void writeEntryWithCustomizedTypeAlsoWritesTypeDeclaration() throws Exception {
     }
 
     @Test
-    void roundtrip() throws Exception {
+    void writeEntryWithConstantMonthApril() throws Exception {
+        BibEntry entry = new BibEntry.Builder(StandardEntryType.Misc)
+                .field(StandardField.MONTH, "#apr#")
+                .build();
+        database.insertEntry(entry);
+
+        databaseWriter.savePartOfDatabase(bibtexContext, Collections.singletonList(entry));
+
+        assertEquals(OS.NEWLINE +
+                "@Misc{," + OS.NEWLINE +
+                "  month = apr," + OS.NEWLINE +
+                "}" + OS.NEWLINE + OS.NEWLINE +
+                "@Comment{jabref-meta: databaseType:bibtex;}" + OS.NEWLINE, stringWriter.toString());
+    }
+
+    @Test
+    void writeEntryWithMonthApril() throws Exception {
+        BibEntry entry = new BibEntry.Builder(StandardEntryType.Misc)
+                .field(StandardField.MONTH, "apr")
+                .build();
+        database.insertEntry(entry);
+
+        databaseWriter.savePartOfDatabase(bibtexContext, Collections.singletonList(entry));
+
+        assertEquals(OS.NEWLINE +
+                "@Misc{," + OS.NEWLINE +
+                "  month = {apr}," + OS.NEWLINE +
+                "}" + OS.NEWLINE + OS.NEWLINE +
+                "@Comment{jabref-meta: databaseType:bibtex;}" + OS.NEWLINE, stringWriter.toString());
+    }
+
+    @Test
+    void roundtripWithArticleMonths() throws Exception {
+        Path testBibtexFile = Paths.get("src/test/resources/testbib/articleWithMonths.bib");
+        Charset encoding = StandardCharsets.UTF_8;
+        ParserResult result = new BibtexParser(importFormatPreferences, fileMonitor).parse(Importer.getReader(testBibtexFile, encoding));
+
+        when(preferences.getEncoding()).thenReturn(encoding);
+        when(preferences.isSaveInOriginalOrder()).thenReturn(true);
+        BibDatabaseContext context = new BibDatabaseContext(result.getDatabase(), result.getMetaData(),
+                new Defaults(BibDatabaseMode.BIBTEX));
+
+        databaseWriter.savePartOfDatabase(context, result.getDatabase().getEntries());
+        try (Scanner scanner = new Scanner(testBibtexFile, encoding.name())) {
+            assertEquals(scanner.useDelimiter("\\A").next(), stringWriter.toString());
```

Comment:
use `Files.readAllLines` instead of the `Scanner`?

Reply:
I tried `Files.readString`. Failed, because of newline issues (LF vs. CRLF).

I came up with

    Files.readAllLines(testBibtexFile, Charset.forName(encoding.name())).stream().collect(Collectors.joining(OS.NEWLINE)) + OS.NEWLINE

Feels more complicated and I can't really get it to work as the assertion fails, but "Contents are equal"